src/window.mm

```diff
@@ -148,6 +148,40 @@ - (void)notified:(const std::string &)note withData:(const msgpack::object &)upd
         AppDelegate *app = (AppDelegate *)[NSApp delegate];
         [app newWindow];
     }
+    else if (note == "neovim.app.openFile") {
+        NSArray *files = [mMainView showFileOpenDialog];
+        if (files != nil) {
+            std::vector<char *> args;
+
+            /* open in tabs if more than one file was selected */
+            if ([files count] > 1)
+                args.push_back(const_cast<char *>("-p"));
+
+            for (NSURL *url in files) 
+                args.push_back(const_cast<char *>([[url path] UTF8String]));
+
+            AppDelegate *app = (AppDelegate *)[NSApp delegate];
+            [app newWindowWithArgs:args];
```

Comment:
@bambu @rogual I guess the default should be opening a new window. It just makes sense as for similar editors. The working directory can be set for each window separately then, which is very convenient. Opening a new file/folder would be like opening a project and once your working directory is set you'll probably open files through vim anyway